components/service/fxa/src/main/java/mozilla/components/fxa/FirefoxAccount.kt

```diff
@@ -0,0 +1,133 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.fxa
+
+import android.text.TextUtils
+import android.util.Log
+
+class FirefoxAccount(override var rawPointer: FxaClient.RawFxAccount?) : RustObject<FxaClient.RawFxAccount>() {
+
+    constructor(config: Config, clientId: String): this(null) {
+        val e = Error.ByReference()
+        val result = FxaClient.INSTANCE.fxa_new(config.consumePointer(), clientId, e)
+        if (e.isSuccess()) {
+            this.rawPointer = result
+        } else {
+            Log.e("FirefoxAccount.init", e.consumeMessage())
+            this.rawPointer = null
+        }
+    }
+
+    override fun destroyPointer(p: FxaClient.RawFxAccount) {
+        FxaClient.INSTANCE.fxa_free(p)
+    }
+
+    fun beginOAuthFlow(redirectURI: String, scopes: Array<String>, wantsKeys: Boolean): String? {
+        val scope = TextUtils.join(" ", scopes)
```

Comment:
nit: TextUtils is not needed in Kotlin: [joinToString](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/join-to-string.html)

Reply:
Fixed.